src/tasks.coffee

```diff
@@ -117,9 +118,11 @@ processNextTaskRound = (task, callback) ->
 
       promises.push defer.promise
 
-  task.save ->
```

Comment:
Arg, it appears that the two saves compete against each other as they are so close together. Sometimes the the tasks get stuck as processing. I've only seen this on travis as it works fine on my machine. Maybe this is becoming evident cos travis is slower. Do you think it is ok to just save once they have been updated?